docs/content/docs/build/tokenproviders.md

```diff
@@ -0,0 +1,150 @@
+
+```typescript
+private async getToken(tenantId: string, documentId: string): Promise<string> {
+    const params = {
+        tenantId,
+        documentId,
+        userId: this.user?.userId,
+        userName: this.user?.userName,
+    };
+    const token = this.getTokenFromServer(params);
+    return token;
+}
+```
+
+The example below uses the [axios](https://www.npmjs.com/package/axios) library to make HTTP requests. You can use other libraries or approaches to making an HTTP request.
+
+```typescript
+private async getTokenFromServer(input: any): Promise<string> {
+    return axios.get(this.azFunctionUrl, {
+        params: input,
+    }).then((response) => {
+        return response.data as string;
+    }).catch((err) => {
+        return err as string;
+    });
+}
+```
+
+The below code snippet can help you with creating your own `HTTPTrigger Azure Function` for fetching the token by passing in your tenant key.
+
+```typescript
+import { AzureFunction, Context, HttpRequest } from "@azure/functions";
+import { ScopeType } from "@fluidframework/azure-client";
+import { generateToken } from "@fluidframework/azure-service-utils";
+import { generateUser } from "@fluidframework/server-service-utils";
+
+//Replace "myTenantKey" with your key here.
+const key = "myTenantKey";
+
+const httpTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
+    // tenantId and documentId are required parameters
+    const tenantId = (req.query.tenantId || (req.body && req.body.tenantId)) as string;
+    const documentId = (req.query.documentId || (req.body && req.body.documentId)) as string;
+    // userId, userName, and scopes are optional and will be filled with default values if not provided
+    const userId = (req.query.userId || (req.body && req.body.userId)) as string;
+    const userName = (req.query.userName || (req.body && req.body.userName)) as string;
+    const scopes = (req.query.scopes || (req.body && req.body.scopes)) as ScopeType[];
+
+    if (!tenantId) {
+        context.res = {
+            status: 400,
+            body: "No tenantId provided in query params",
+        };
+    }
+
+    if (!key) {
+        context.res = {
+            status: 404,
+            body: `No key found for the provided tenantId: ${tenantId}`,
+        };
+    }
+
+    if (!documentId) {
+        context.res = {
+            status: 400,
+            body: "No documentId provided in query params"
+        };
+    }
+
+    let user = { name: userName, id: userId };
+    if (!userId || !userName) {
+        // generateUser will generate a random userName and userId. If either of these values are not provided as
+        // query parameters, the randomly generated values will be used instead.
+        const generatedUser = generateUser() as any;
```

Comment:
I think we should drop the random user generation here and instead expect people to pass in the user object for this production scenario. This will also allow us to avoid having the `@fluidframework/server-service-utils` dependency